boring/src/ssl/mod.rs

```diff
@@ -1958,15 +1987,19 @@ impl ClientHello {
             let mut ptr = ptr::null();
             let mut len = 0;
             let result =
-                ffi::SSL_early_callback_ctx_extension_get(&self.0, ext_type.0, &mut ptr, &mut len);
+                ffi::SSL_early_callback_ctx_extension_get(self.0, ext_type.0, &mut ptr, &mut len);
             if result == 0 {
                 return None;
             }
             Some(slice::from_raw_parts(ptr, len))
         }
     }
 
-    fn ssl(&self) -> &SslRef {
+    pub fn ssl_mut(&mut self) -> &mut SslRef {
+        unsafe { SslRef::from_ptr_mut(self.0.ssl) }
+    }
+
+    pub fn ssl(&self) -> &SslRef {
```

Comment:
Dangling pointer?

Reply:
Not sure what you mean with this question.

Reply:
Nevermind, I misunderstood what this was doing internally. I didn't realize that the `from_ptr_mut` function returned `&'a mut Self`.

I assumed that this was a newtype over a smart pointer, and so I thought that a dangling ref to a smart pointer was being returned.

I see what you're doing here now, I agree that this is fine.